src/index.ts

```diff
@@ -8,32 +8,100 @@ import {
 
 const client = new CloudWatchClient();
 
+const VALID_STATUSES = new Set(["attempt", "success", "failure"]);
+
 program
   .name("deployment-stats")
   .description("CLI to push statsd metrics to CloudWatch");
 
 program
   .command("deployment")
   .argument("<project>")
-  .option("-s, --status <status>")
-  .option("-e, --environment <environment>")
+  .option("-s, --status <status>", "Deployment status")
+  .option("-e, --environment <environment>", "Deployment environment")
+  .option("-d --deploy-duration <deploy-duration>", "Deployment duration")
   .action(async (project, options) => {
-    const input: PutMetricDataCommandInput = {
-      MetricData: [
-        {
-          MetricName: `deployment.${options.status}`,
-          Dimensions: [
-            { Name: "Environment", Value: options.environment },
-            { Name: "Project", Value: project },
-          ],
-          Unit: "Count",
-          Value: 1,
-        },
-      ],
-      Namespace: "cuckoo",
-    };
-    await client.send(new PutMetricDataCommand(input));
-    console.log("Metric published to CloudWatch");
+    try {
+      const deployDuration = Number.parseInt(options.deployDuration);
+      if (Number.isNaN(deployDuration)) {
+        throw new TypeError("Deploy duration must be a number");
+      }
+
+      if (!VALID_STATUSES.has(options.status)) {
+        throw new TypeError("Invalid status");
+      }
+
+      const input: PutMetricDataCommandInput = {
+        MetricData: [
+          {
+            MetricName: `deployment.${options.status}`,
+            Dimensions: [
+              { Name: "Environment", Value: options.environment },
+              { Name: "Project", Value: project },
+              { Name: "Status", Value: options.status },
```

Comment:
You probably don't need a dimension for status on its own as the status is already included in the metric name